### PR TITLE
Deeplink encoding

### LINKED ISSUE #3: Add deeplink encoding
**Is your feature request related to a problem? Please describe.**

### Scenario
You want to share artist's event.

### Details
You have a deeplink 

```swift
"artist/\(\.id)/event/\(\.eventId)" as Deeplink<ArtistEvent>
```

Instead of remember parameters order, you can invoke method `encode`:

```swift
let artistEvent = ArtistEvent(
  id: "123",
  event: "456"
)
deeplink.encode(artistEvent) // return "artist/123/event/456"
```

**Describe the solution you'd like**

Add method `encode` to `Deeplink<Value>` and `DeeplinkInterpolation<Value>`.

### PR DESCRIPTION
**Pull request by lolgear**
https://github.com/TizianoCoroneo/Deeplink/issues/3

Closes #3

### CODE REVIEW

File: Sources/Deeplink/Deeplink.swift [hunk 1]
```diff
@@ -45,6 +45,12 @@ public enum Deeplink<Value>: Equatable, Hashable {
         case .interpolated(let interpolation): return interpolation
         }
     }
+
+    public func encode(
```
Comment (TizianoCoroneo) [inline Sources/Deeplink/Deeplink.swift:69]: Can you please add a documentation comment to this line? It doesn't matter for the internals of this feature, but I try to keep all public API documented

**Review by TizianoCoroneo — APPROVED** [review]
Hello, thanks for the contribution! I like the improvement, I just have a small comment

File: Sources/Deeplink/Deeplink.swift (hunk 1 above)
Comment (lolgear) [reply Sources/Deeplink/Deeplink.swift:69]: Ok, can you check it?

Comment (TizianoCoroneo) [reply Sources/Deeplink/Deeplink.swift:69]: Thanks!

**Review by TizianoCoroneo — APPROVED** [review]

**Pull request merged**